minigrid/envs/wfc/graphtransforms.py

```diff
@@ -0,0 +1,404 @@
+from __future__ import annotations
+
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from itertools import product
+
+import networkx as nx
+import numpy as np
+
+from minigrid.core.constants import COLOR_TO_IDX, IDX_TO_OBJECT, OBJECT_TO_IDX
+from minigrid.minigrid_env import MiniGridEnv
+
+
+@dataclass
+class EdgeDescriptor:
+    between: tuple[str, str] | tuple[str]
+    structure: str | None = None
+    weight = None
```

Comment:
Could we add a data type

Reply:
Do you mean a special type for the `tuple[str, str]` type?

Reply:
Every other attribute in `EdgeDescriptor` has a type other than `weight`